superset/commands/dataset/create.py

```diff
@@ -85,10 +88,12 @@ def validate(self) -> None:
             exceptions.append(TableNotFoundValidationError(table_name))
 
         if sql:
+            processor = jinja_context.get_template_processor(database=database)
```

Comment:
@betodealmeida I think we need this in more places than just virtual datasets per my comment in the PR, i.e., a chart (via an ad-hoc filter), SQL Lab query, etc. could contain a Jinja template.

Should this two-phase approach be part of the `ParsedQuery` class so it's applicable to all use cases?

Reply:
@john-bodley as far as I could tell, in those places it's already handled correctly. For SQL Lab I can confirm that the template is processed before being passed to the security manager for parsing. I'll test with ad-hoc filters, but when I inspected the code I only saw this place in need of fixing.